web-app-router/scenes/Portal/Teams/TeamId/Team/page/Members/List/RemoveUserDialog/index.tsx

```diff
@@ -0,0 +1,101 @@
+"use client";
+
+import { CircleIconContainer } from "@/components/CircleIconContainer";
+import { DecoratedButton } from "@/components/DecoratedButton";
+import { Dialog, DialogProps } from "@/components/Dialog";
+import { DialogOverlay } from "@/components/DialogOverlay";
+import { DialogPanel } from "@/components/DialogPanel";
+import { AlertIcon } from "@/components/Icons/AlertIcon";
+import { TYPOGRAPHY, Typography } from "@/components/Typography";
+import { atom, useAtom } from "jotai";
+import { useCallback } from "react";
+import { useForm } from "react-hook-form";
+import { useRemoveUserMutation } from "./graphql/client/remove-user.generated";
+import { useParams } from "next/navigation";
+import { toast } from "react-toastify";
+import { FetchMembershipsDocument } from "../graphql/client/fetch-members.generated";
+
+export const removeUserDialogAtom = atom(false);
+
+export const RemoveUserDialog = (props: {
+  name: string;
+  id: string | undefined | null;
+}) => {
+  const [isOpened, setIsOpened] = useAtom(removeUserDialogAtom);
+  const { teamId } = useParams() as { teamId: string };
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = useForm();
+
+  const [removeUser] = useRemoveUserMutation({
+    context: { headers: { team_id: teamId } },
+  });
+
+  const submit = useCallback(async () => {
+    if (!props.id || !teamId) {
+      return toast.error("Something went wrong. Please try again later.");
+    }
+
+    try {
+      await removeUser({
+        variables: {
+          teamId,
+          userId: props.id,
+        },
+
+        refetchQueries: [FetchMembershipsDocument],
```

Comment:
might be worth awaiting the refetch in case the query takes longer and the dialog closes, but the membership list doesn't refetch

```suggestion
        refetchQueries: [FetchMembershipsDocument],
        awaitRefetchQueries: true,
```